### DISCUSSION TITLE
Purpose of cloneSubmodules config option

### DISCUSSION THREAD

**Discussion by jjathman** [renovatebot/renovate#8924] [Request Help]
We are getting a warning of `Unable to initialise git submodule at [location]`. Since we don't really care about that submodule being processed by Renovate we added the configuration option of `cloneSubmodules: false` in the repository, but we are still getting that error message. Does the repository configuration get processed after that submodule warning is logged?

Secondarily, I think it would be nice if that warning included some details about why the submodule couldn't be initialized, it might be easier for us to just resolve that problem but I don't have any details to go off of why it couldn't be initialized.

Thank you!

**Comment by rarkins** [comment]
`cloneSubmodules` is passed to the git layer as part of the `platform.initRepo()`, and that seems to be something we do before fetching the repo config. If so then that would explain why it's not working if you put it in your `renovate.json`.

@JamieMagee this option is [listed in user config options](https://docs.renovatebot.com/configuration-options/#clonesubmodules) but seems like it's not possible to use unless in the admin config. I don't see any issues open about it yet, but maybe we treat this as a bug? https://github.com/renovatebot/renovate/search?q=cloneSubmodules+is%3Aissue&type=issues

**Reply by jjathman** [reply]
Thanks, it's helpful to know that I'm not missing something obvious. Is there another way to accomplish what I'm trying to do? Or a way to see what the error is that's preventing it from being initialized?

**Reply by JamieMagee** [reply]
Yeah, I should update the configuration for that option, and add some more logging there.

@jjathman until then, can you set the log level higher. That might get you some more insight.

**Reply by jjathman** [reply]
I don't think there's any more information to see. Could we get the error message to be printed along with this warning?

https://github.com/renovatebot/renovate/blob/093f3d262ad3926fed91d7aaf4a5ccb0399b6f8e/lib/util/git/index.ts#L299-L300

**Reply by rarkins** [reply]
I've added the `err` to the warn log, it should propagate up to the app tomorrow
